spark/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -178,6 +195,30 @@ public StructType readSchema() {
     return lazyType();
   }
 
+  /**
+   * This is called in the Spark Driver when data is to be materialized into {@link ColumnarBatch}
+   */
+  @Override
+  public List<InputPartition<ColumnarBatch>> planBatchInputPartitions() {
```

Comment:
I don't see a test for this. Can you update some of the Spark tests to run both vectorized and non-vectorized? We can also do this in a follow-up, but we need to make sure that this code path is being tested as thoroughly as the non-vectorized code path.

Reply:
I added cases to TestSparkReadProjection that use this path in the PR against your branch.